src/dataframerow/utils.jl

```diff
@@ -440,57 +440,3 @@ function compute_indices(groups::AbstractVector{<:Integer}, ngroups::Integer)
 
     return rperm, starts, stops
 end
-
-# Build RowGroupDict for a given DataFrame, using all of its columns as grouping keys
-function group_rows(df::AbstractDataFrame)
-    groups = Vector{Int}(undef, nrow(df))
-    ngroups, rhashes, gslots, sorted =
-        row_group_slots(ntuple(i -> df[!, i], ncol(df)), Val(true), groups, false, false)
-    rperm, starts, stops = compute_indices(groups, ngroups)
-    return RowGroupDict(df, rhashes, gslots, groups, rperm, starts, stops)
```

Comment:
I think `RowGroupDict` can also be removed.

Reply:
Indeed - I am  running tests to double check. It is astonishing how much we have reworked internally in this release.

Reply:
OK - all seems clean. I will merge the PR after CI passes.

Reply:
Now that this file contains only grouping code, we will be able to move it to the corresponding folder and rename it without touching anything else. :-)

Reply:
OK will move it in this PR as otherwise we will forget. Also `nonunique` uses it, but I think it is not a problem.